engine/runtime/src/main/java/org/enso/interpreter/runtime/data/atom/AtomConstructor.java

```diff
@@ -119,18 +128,17 @@ public AtomConstructor initializeFields(
       Annotation[] annotations,
       ArgumentDefinition... args) {
     CompilerDirectives.transferToInterpreterAndInvalidate();
+    assert boxedLayout == null : "Don't initialize twice: " + this.name;
     if (args.length == 0) {
-      cachedInstance = new BoxingAtom(this);
+      cachedInstance = BoxingAtom.singleton(this);
     } else {
       cachedInstance = null;
     }
-    if (Layout.isAritySupported(args.length)) {
-      boxedLayout = Layout.create(args.length, 0, args);
-    }
+    boxedLayout = Layout.createBoxed(args);
```

Comment:
The major change that fixes #8710:
- each `AtomConstructor` is associated with `boxedLayout`
- we always use `StructsLibrary` (and thus _layouts_) to access the atom values
- as such the `SuspendedFieldGetterNode` is used for `BoxingAtom` as well as `UnboxingAtom`